### PR TITLE
DM-45118: Rewrite blocks and bottlenecks

### PR DESCRIPTION
**Pull request by ugyballoons**
Fixes blocking operations like a very inefficient listcomp and uses asyncronous batch processing for converting, storing and compressing data objects.

The result aims to remove large blocks from the main thread for better app responsiveness.

It also fixes a case where requesting a night report during historical polling would cause a 500 error.

### CODE REVIEW

File: python/lsst/ts/rubintv/handlers/pages.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ async def get_home(
     request: Request,
 ) -> Response:
     """GET ``/rubintv/`` (the app's external root)."""
-    logger.info("Request for the app home page")
+    # logger.info("Request for the app home page")
```
Comment (sebastian-aranda) [inline python/lsst/ts/rubintv/handlers/pages.py:50]: Should this be removed perhaps?

File: python/lsst/ts/rubintv/handlers/pages.py [hunk 2]
```diff
@@ -354,7 +356,7 @@ async def get_historical_night_report_page(
             "location": location,
             "camera": camera.model_dump(),
             "date": day_obs,
-            "night_report": night_report.model_dump(),
+            "night_report": (night_report.model_dump()),
```
Comment (sebastian-aranda) [inline python/lsst/ts/rubintv/handlers/pages.py:359]: I think this parenthesis is not necessary.

File: python/lsst/ts/rubintv/background/historicaldata.py [hunk 3]
```diff
@@ -166,15 +185,16 @@ async def store_events(self, events: list[Event], locname: str) -> None:
                 self._calendar[loc_cam][year][month][day] = seq_num
         logger.info("ending store_events")
 
-    async def storage_name_for_event(self, event: Event, locname: str) -> str:
+    async def storage_key_for_event(self, event: Event, locname: str) -> str:
```
Comment (sebastian-aranda) [inline python/lsst/ts/rubintv/background/historicaldata.py:188]: I think this function should not be asynchronous.

**Review by sebastian-aranda — CHANGES_REQUESTED** [review]
Nice job @ugyballoons, I've left some minor comments so you can take a look :pray:.

File: python/lsst/ts/rubintv/background/historicaldata.py (hunk 3 above)
Comment (ugyballoons) [reply python/lsst/ts/rubintv/background/historicaldata.py:188]: Agreed!

**Review by sebastian-aranda — APPROVED** [review]
Thanks for the good work on this @ugyballoons ! This looks good to me :+1:.

**Pull request merged**